app/common/state/bookmarksState.js

```diff
@@ -187,7 +187,9 @@ const bookmarksState = {
     }
 
     siteKeys.forEach((siteKey) => {
-      state = state.setIn([STATE_SITES.BOOKMARKS, siteKey, 'favicon'], favicon)
+      if (state.getIn([STATE_SITES.BOOKMARKS, siteKey])) {
```

Comment:
If we only checking here if exists I would change it to `state.hasIn([STATE_SITES.BOOKMARKS, siteKey])`